### PR TITLE
added backend test image + publishing

### PR DESCRIPTION
**Pull request by mzedel**
this is to be shared between all the acceptance tests

@alfrunes this is what I meant in https://github.com/mendersoftware/deviceauth/pull/408 :)

Changelog: None
Signed-off-by: Manuel Zedel <manuel.zedel@northern.tech>

### CODE REVIEW

**Comment by lluiscampos** [comment]
Great initiative :tada: 

I think we can discuss it a little before jumping into the implementation, though. I originally thought of having it in `mendertesting` repo to be closer to the template. Here is also an option, maybe in a subfolder.

**Comment by mzedel** [comment]
That was bothering me since I first saw the first duplicate 😄...
Based on the _Contains shared code for creating containers for test purposes_ description I thought this repo was most fitting, but I'll be happy to relocate - for now I put the 2 files in a subfolder...

I'm also slightly abusing the docker repo and tagging for this 🤡 - considering the absence of versioning for these I thought that might be tolerable in the face of creating another docker image...

**Comment by lluiscampos** [comment]
@mzedel Okey, let's get it in this repo then.

With regards to the image name, I think it should be something more explicit than `testting`, as this repo also builds images for raspbian testing. What about `acceptance-tester` or `acceptance-testing`...?

**Comment by mzedel** [comment]
ah of course... I just reused the image name from the acceptance tests 🙈 and ignored the context here... but with this we could give this approach a try and see how many repos can get along with the image - what do you think @lluiscampos?

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -49,10 +49,25 @@ build:
...
 
+build:testing:
+  stage: build
+  dependencies:
+    - build_prep
+  services:
+    - docker:dind
+  script:
+    - docker build -t $DOCKER_REPOSITORY:testing -f backend-acceptance-testing/Dockerfile.backend-tests backend-acceptance-testing
+    - docker save $DOCKER_REPOSITORY:testing > testingImage.tar
```
Comment (lluiscampos) [inline .gitlab-ci.yml:60]: Here also:

```suggestion
    - docker save $DOCKER_REPOSITORY:acceptance-testing > testingImage.tar
```

**Review by lluiscampos — CHANGES_REQUESTED** [review]

**Comment by lluiscampos** [comment]
@mzedel You should have rights now, sorry for that.

File: .gitlab-ci.yml (hunk 1 above)
Comment (mzedel) [reply .gitlab-ci.yml]: argh, sorry for the oversights... should be aligned now!

**Review by lluiscampos — APPROVED** [review]

**Pull request merged**